### PR TITLE
[FIXED JENKINS-27441] avoid IOOBE in getLog when called with 0

### PR DESCRIPTION
**Pull request by vb-linetco**
return an empty list when called with maxLines = 0 instead

### CODE REVIEW

**Comment by olivergondza** [comment]
+1, can you add a unit test?

**Comment by olivergondza** [comment]
Unittest does not reproduce the problem:
```
getLogReturnsAnEmptyListWhenCalledWith0(hudson.model.RunTest)  Time elapsed: 0.449 sec  <<< ERROR!
java.io.FileNotFoundException: .../jenkins/core/target/0/log (No such file or directory)
	at java.io.FileInputStream.open(Native Method)
	at java.io.FileInputStream.<init>(FileInputStream.java:146)
	at hudson.model.Run.getLog(Run.java:1942)
	at hudson.model.RunTest.getLogReturnsAnEmptyListWhenCalledWith0(RunTest.java:128)
```

When I create the log file before trying to read it, it succeeds. What is the complete stacktrace of the problem you are fixing? Also, consider creating JIRA isues with all the detail for bugs.

**Comment by vb-linetco** [comment]
I've created an issue in JIRA, i think you have to add at least one line to the log to trigger the exception.
I've no clue how to add some log lines in the test case. Any pointers how I can add some log entries would be appreciated.

**Comment by olivergondza** [comment]
> Any pointers how I can add some log entries would be appreciated.

IIRC `org.apache.commons.io.FileUtils` provides such helper functions.

**Comment by daniel-beck** [comment]
I'd prefer a squashed PR. 9 commits for this change is a bit much IMO.

Renaming PR so a merge will automatically close the issue (https://wiki.jenkins-ci.org/display/JENKINS/contributing#contributing-CommitMessages)…

**Comment by bootstraponline** [comment]
I ran into this issue today on cloudbees. It'd be nice to get this fixed.

**Comment by daniel-beck** [comment]
I missed that the PR was updated to use a single commit. Sorry about that!

Cannot be merged cleanly anymore, so please update the PR.

**Comment by vb-linetco** [comment]
Updated the PR, can be automatically merged again.

**Comment by bootstraponline** [comment]
I think you need to do:

```
git pull --rebase git@github.com:jenkinsci/jenkins.git
git push -f
```

This PR should show exactly one commit, not 43.

**Comment by vb-linetco** [comment]
Thx for hint, forgot to rebase. Should be one commit now.

**Comment by bootstraponline** [comment]
Looks great. Thanks!

**Comment by oleg-nenashev** [comment]
LGTM

File: core/src/test/java/hudson/model/RunTest.java [hunk 1]
```diff
@@ -128,4 +131,18 @@ public void getDurationString() throws IOException {
...
+    public void getLogReturnsAnEmptyListWhenCalledWith0() throws Exception {
+        Run r = new Run(new StubJob(), 0) {};
+        File f = r.getLogFile();
+        f.getParentFile().mkdirs();
+        PrintWriter w = new PrintWriter(f, "utf-8");
+        w.println("dummy");
+        w.close();
+        List<String> logLines = r.getLog(0);
+        assertTrue(logLines.isEmpty());
+    }
```
Comment (jtnord) [inline core/src/test/java/hudson/model/RunTest.java]: the file is never deleted AFAICT and will litter the temp directory.

Comment (hrzlgnm) [inline core/src/test/java/hudson/model/RunTest.java]: should not be littering anymore

Comment (vb-linetco) [inline core/src/test/java/hudson/model/RunTest.java]: should not be littering in temp directory anymore

**Comment by jtnord** [comment]
:+1:

**Comment by oleg-nenashev** [comment]
@vb-linetco 
Merged, thanks for the fix!

**Pull request merged**